nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -167,10 +167,6 @@ exec 1>&$logOutFd 2>&$logErrFd
 exec {logOutFd}>&- {logErrFd}>&-
 
 
-# Start systemd.
+# Start systemd in a clean environment.
 echo "starting systemd..."
-
-PATH=/run/current-system/systemd/lib/systemd:@fsPackagesPath@ \
```

Comment:
This will remove `/run/current-system/systemd/lib/systemd` from systemd's `$PATH`.

Isn't that used to pick up some of the systemd-internal tooling?

`systemd.service` manpage, `ExecStart` section says:
```
           For each of the specified commands, the first argument must be either an absolute path to an executable or a simple file name without any slashes. Optionally,
           this filename may be prefixed with a number of special characters:
```

We do have some systemd-provided units with a non-absolute path, for example `systemctl cat systemd-tmpfiles-setup.service`. Do they fail? Or is another path baked in here?

Reply:
No, I ran the test and it seems to use the compiled-in value

Reply:
What should the comment say? "Sets $PATH for systemd"? I have really no idea what the point is here

Reply:
https://github.com/NixOS/nixpkgs/pull/165453/files#diff-5ea4693beb4bdd8f7efcdd3204ceea67c86f59b3d1c28cde063ddf03e1d20fbfR559 is fine :+1: